### PR TITLE
Added checks for side to for PricingError and checking for use_order_…

### PR DESCRIPTION
**Pull request by samgermain**
## Summary
Solve the issue: #5334 

## Quick changelog

In `exchange.get_rate`

```
if conf_strategy.get('use_order_book', False) and ('use_order_book' in conf_strategy):
```

was changed to

```
dont_use_order_book = (
        ('use_order_book' in conf_strategy or side == "sell") and
        conf_strategy.get('use_order_book', False)
)

if dont_use_order_book:
```

and

```
if rate is None:
            raise PricingError(f"{name}-Rate for {pair} was empty.")
```

was changed to 

```
if rate is None and side == "sell":
            raise PricingError(f"Sell-Rate for {pair} was empty.")
 ```

## What's new?
In PR #5284 these changes were made when merging the 2 methods

<table style="width:100%">
  <tr>
    <th>get_buy_rate</th>
    <th>get_sell_rate</th> 
    <th>update</th>
  </tr>
  <tr>
    <td>if bid_strategy.get('use_order_book', False) and ('use_order_book' in bid_strategy):</td>
    <td>if ask_strategy.get('use_order_book', False)</td>
    <td>if strategy.get('use_order_book', False) and ('use_order_book' in strategy) is now used for both</td>
  </tr>
  <tr>
     <td>/</td>
     <td>if rate is None: raise PricingError(f"{name}-Rate for {pair} was empty.")</td>
     <td>Both raise this pricing error now</td>
  </tr>
</table>

Both changes are reverted in this PR. I'm guessing that reverting the first change will probably fix the issue  #5334 
I reverted the second change because I assume that the original `get_buy_rate` strategy didn't check for this for a reason given that the other changes have caused problems

### CODE REVIEW

**Comment by samgermain** [comment]
I'm trying to find a strategy that will quickly make a buy and a sell in live mode so I can test out that everything is working fine for sure

**Comment by samgermain** [comment]
I think I got blocked by binance for sending too many requests.  I was on the 1m timeframe and had "max_open_trades": 1,

File: freqtrade/exchange/exchange.py
Comment (xmatthias) [inline freqtrade/exchange/exchange.py:1061]: Both buy and sell methods did raise a `PricingError` before - and can raise a pricing-error now as well.

As such - i really don't understand this change -n as it's changing behavior over the previous version.

File: freqtrade/exchange/exchange.py
Comment (xmatthias) [inline freqtrade/exchange/exchange.py]: The naming (`dont_use_order_book`) suggests the wrong thing (if `dont_use_order_book=True` - we'll be using orderbook prices) - and as such, is very confusing to even look at.

File: freqtrade/exchange/exchange.py
Comment (xmatthias) [inline freqtrade/exchange/exchange.py]: I'm not sure what you're trying to do with this.
the side has nothing to do with the "use_order_book" setting - it's valid for both ask_strategy and bid_strategy - as such - i don't see why side should make this kind of a difference to this method.



If that were really the case (hint: it isn't), then please revert the whole PR that pulled these 2 methods 
apart.

You can surely simplify `if conf_strategy.get('use_order_book', False) and ('use_order_book' in conf_strategy):` to `if conf_strategy.get('use_order_book', False):` (that 2nd part is irrelevant, as `.get` will use the default if it's not in conf_strategy - but `use_order_book` is perfectly fine on both buy and sell sides.
I understand why it was added the way it was (we had this slight difference between `get_buy_rate` and `get_sell_rate` - while it doesn't hurt - it's also not problematic to not have this 2nd check.

**Review by xmatthias — CHANGES_REQUESTED** [review]
I find basically every changed line here wrong ... 

Getting rates is fine for both sides as is (and `PricingError` was raised before from both sides - and is raised now as well).

the only thing that (slightly) concerns me is that this does not make tests fail - as it should as it's introducing wrong behaviour on the buy side.


On top of that - pricing has nothing to do with #5334.
While the "pricing" message is there - the error is something entirely different.

**Comment by samgermain** [comment]
> I find basically every changed line here wrong ...
> 
> Getting rates is fine for both sides as is (and `PricingError` was raised before from both sides - and is raised now as well).
> 
> the only thing that (slightly) concerns me is that this does not make tests fail - as it should as it's introducing wrong behaviour on the buy side.
> 
> On top of that - pricing has nothing to do with #5334.
> While the "pricing" message is there - the error is something entirely different.

Sounds good, I thought this might have been causing the problems in #5334 

I went to the commit right before the first commit on #5284 ( 03a4ae46746905866f2ab7aba86534d6e0c1791e ) and took  a [screenshot](https://i.imgur.com/vr3QCMn.png) of the `get_buy_rate` and `get_sell_rate` methods and highlighted some stuff, so you can see that the pricing error was only raised in `get_sell_rate` before and the `get_sell_rate` condition didn't check if `use_order_book` was in `ask_strategy`

------------------

Feel free to close the PR. I'll leave it open in case you want it

**Comment by xmatthias** [comment]
I do have been aware of the changes this introduced - and that the methods were not 100% aligned (which is now better though - as the "none" price check was missing on the Buy side (which can happen with duplicated code).

Pricing / Pricing exception was / is however never an issue in the first place with either methods and both methods could raise PricingError (if orderbook is empty / raises Keyerrors, for example) - and the calling code will handle that accordingly.

**Pull request closed without merging**
